power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -36,16 +38,130 @@ static_assert(is_indptr_mutable_v<writable_dataset_t>);
+};
+
+template <typename T, dataset_type_tag dataset_type> class ColumnarAttributeRange {
+  public:
+    using Data = std::conditional_t<is_data_mutable_v<dataset_type>, void, void const>;
+
+    class iterator;
+
+    class Proxy {
+      public:
+        using value_type = std::remove_const_t<T>;
+
+        Proxy() = default;
+        Proxy(Idx idx, std::span<AttributeBuffer<Data> const> attribute_buffers)
+            : idx_{idx}, attribute_buffers_{std::move(attribute_buffers)} {}
+
+        Proxy& operator=(value_type const& value)
+            requires is_data_mutable_v<dataset_type>
+        {
+            for (auto const& attribute_buffer : attribute_buffers_) {
+                assert(attribute_buffer.meta_attribute != nullptr);
+                auto const& meta_attribute = *attribute_buffer.meta_attribute;
+                ctype_func_selector(
+                    meta_attribute.ctype, [&value, &attribute_buffer, &meta_attribute, this]<typename AttributeType> {
+                        AttributeType* buffer_ptr = reinterpret_cast<AttributeType*>(attribute_buffer.data) + idx_;
+                        AttributeType const& attribute_ref = meta_attribute.template get_attribute<AttributeType const>(
+                            reinterpret_cast<RawDataConstPtr>(&value));
+                        *buffer_ptr = attribute_ref;
+                    });
+            }
+            return *this;
+        }
+        operator value_type() const { return get(); }
+        value_type get() const {
+            value_type result{};
+            for (auto const& attribute_buffer : attribute_buffers_) {
+                assert(attribute_buffer.meta_attribute != nullptr);
+                auto const& meta_attribute = *attribute_buffer.meta_attribute;
+                ctype_func_selector(
+                    meta_attribute.ctype, [&result, &attribute_buffer, &meta_attribute, this]<typename AttributeType> {
+                        AttributeType const* buffer_ptr =
+                            reinterpret_cast<AttributeType const*>(attribute_buffer.data) + idx_;
+                        AttributeType& attribute_ref =
+                            meta_attribute.template get_attribute<AttributeType>(reinterpret_cast<RawDataPtr>(&result));
+                        attribute_ref = *buffer_ptr;
+                    });
+            }
+            return result;
+        }
+
+      private:
+        friend class ColumnarAttributeRange;
+        friend class iterator;
+
+        Idx idx_{};
+        std::span<AttributeBuffer<Data> const> attribute_buffers_{};
+    };
+
+    class iterator : public boost::iterator_facade<iterator, T, boost::random_access_traversal_tag, Proxy, Idx> {
+      public:
+        using value_type = Proxy;
+
+        iterator() = default;
+        iterator(Idx idx, std::span<AttributeBuffer<Data> const> attribute_buffers)
+            : idx_{idx}, current_{idx, attribute_buffers} {}
+
+        Idx get_idx() const { return idx_; }
```

Comment:
why do you need this function?

Reply:
It was used in old (now obsolete) implementation. I will remove now.